### ISSUE TITLE
`SoundEffect` with frequencies above 10 KHz don't make a sound

### ISSUE THREAD

**Issue by microbit-carlos** [microbit-foundation/micropython-microbit-v2#120]
I'm guessing the frequency string format only takes 4 digits?

This works:
```python
audio.play(audio.SoundEffect(
    freq_start=9999,
    freq_end=9999,
    duration=300,
    vol_start=255,
    vol_end=255,
))
```

But this doesn't make a sound:
```python
audio.play(audio.SoundEffect(
    freq_start=10000,
    freq_end=10000,
    duration=300,
    vol_start=255,
    vol_end=255,
))
```

I think `10000` is likely converted to `0`. As a SoundEfect like this sounds like it's going from high frequency to low frequency.
```python
audio.play(audio.SoundEffect(
    freq_start=9999,
    freq_end=10000,
    duration=500,
    vol_start=255,
    vol_end=255,
))
```

Should we throw an exception or clamp for values >10K?

**Comment by microbit-carlos** [comment]
I haven't tried but I guess the same would be true for volume and duration.

**Comment by dpgeorge** [comment]
> I'm guessing the frequency string format only takes 4 digits?

Yes that's correct.  Most of the numbers (volume, frequency, duration) are limited to 4 digits.

> Should we throw an exception or clamp for values >10K?

Implementing either of those is easy.  Probably best to raise an exception.

**Comment by microbit-carlos** [comment]
> Probably best to raise an exception.

Sounds good 👍

**Comment by dpgeorge** [comment]
Fixed by 0b7552643cb0ab78e511b0772beca2fe3b970833
